### PR TITLE
Fix arcadia src/Access gtest 

### PR DESCRIPTION
**Pull request by qoega**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):

- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

Detailed description / Documentation draft:
...

> By adding documentation, you'll allow users to try your new feature immediately, not when someone else will have time to document it later. Documentation is necessary for all features that affect user experience in any way. You can add brief documentation draft above, or add documentation right into your patch as Markdown files in [docs](https://github.com/ClickHouse/ClickHouse/tree/master/docs) folder.

> If you are doing this for the first time, it's recommended to read the lightweight [Contributing to ClickHouse Documentation](https://github.com/ClickHouse/ClickHouse/tree/master/docs/README.md) guide first.

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Storages/MergeTree/IMergeTreeReader.cpp [hunk 1]
```diff
@@ -50,7 +50,7 @@ IMergeTreeReader::IMergeTreeReader(
 
     columns_from_part.set_empty_key(StringRef());
     for (const auto & column_from_part : part_columns)
-        columns_from_part.emplace(column_from_part.name, &column_from_part.type);
```
Comment (alexey-milovidov) [inline src/Storages/MergeTree/IMergeTreeReader.cpp:53]: The semantic is different. Emplace does not change the value if it exists.

File: src/Storages/StorageInMemoryMetadata.cpp [hunk 2]
```diff
@@ -331,7 +331,7 @@ Block StorageInMemoryMetadata::getSampleBlockForColumns(
     /// Virtual columns must be appended after ordinary, because user can
     /// override them.
     for (const auto & column : virtuals)
-        virtuals_map.emplace(column.name, &column.type);
```
Comment (alexey-milovidov) [inline src/Storages/StorageInMemoryMetadata.cpp:334]: The semantic is changed, is it what you expect?

Comment (qoega) [reply src/Storages/StorageInMemoryMetadata.cpp:334]: I just applied patch from @CurtisJ here

**Comment by alexey-milovidov** [comment]
Nevertheless it does not work :(

File: src/Common/DenseHashMap.h [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#pragma once
+#include <unordered_map>
+
+#if defined(ARCADIA_BUILD)
```
Comment (alexey-milovidov) [inline src/Common/DenseHashMap.h:4]: This is extraordinarily ugly.

Need to add comments with a short essay about "What is Arcadia and why do we need this uglyness".

File: src/Core/NamesAndTypes.cpp [hunk 4]
```diff
@@ -6,7 +6,7 @@
 #include <IO/WriteHelpers.h>
 #include <IO/ReadBufferFromString.h>
 #include <IO/WriteBufferFromString.h>
-#include <sparsehash/dense_hash_map>
```
Comment (alexey-milovidov) [inline src/Core/NamesAndTypes.cpp:9]: Maybe reusing our `HashMap` (by just providing proper template parameters) will be easier.

File: src/Common/DenseHashMap.h (hunk 3 above)
Comment (alexey-milovidov) [reply src/Common/DenseHashMap.h:4]: Other code LGTM.

Comment (KochetovNicolai) [reply src/Common/DenseHashMap.h:4]: > This is extraordinarily ugly.

Agree

**Comment by KochetovNicolai** [comment]
The last error in arcadia is
```
ld: error: undefined symbol: DB::registerFunctionTupleToNameValuePairs(DB::FunctionFactory&)
>>> referenced by registerFunctionsTuple.cpp:14 (/-S/clickhouse/src/Functions/registerFunctionsTuple.cpp:14)
>>>               registerFunctionsTuple.cpp.o:(DB::registerFunctionsTuple(DB::FunctionFactory&)) in archive clickhouse/src/Functions/libclickhouse-src-Functions.a
clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
```

I am tired of updating ya.make - going to merge

**Pull request merged**